src/nvme/types.h

```diff
@@ -9007,4 +9025,350 @@ struct nvme_ns_mgmt_host_sw_specified {
 };
 #endif /* SWIG */
 
+/**
+ * enum nvme_lm_cdq_fields - Controller Data Queue command fields
+ *
+ * @NVME_LM_CREATE_CDQ:					Create CDQ select option
+ * @NVME_LM_DELETE_CDQ:					Delete CDQ select option
+ * @NVME_LM_CDQ_SELECT_SHIFT:				Shift to set Select (SEL) field
+ * @NVME_LM_CDQ_SELECT_MASK:				Mask to set SEL field
+ * @NVME_LM_QUEUE_TYPE_USER_DATA_MIGRATION_QUEUE:	User Data Migration Queue type
+ * @NVME_LM_QUEUE_TYPE_SHIFT:				Shift amount to set Queue Type (QT) field
+ * @NVME_LM_QUEUE_TYPE_MASK:				Mask to set QT field
+ * @NVME_LM_CREATE_CDQ_PC:				Physically Contiguous
+ * @NVME_LM_CREATE_CDQ_CNTLID_SHIFT:			Shift amount to set CNTLID field
+ * @NVME_LM_CREATE_CDQ_CNTLID_MASK:			Mask to set CNTLID field
+ * @NVME_LM_CREATE_CDQ_CDQID_SHIFT:			Shift amount to get CDQID field from Create
+ *							response in completion dword0
+ * @NVME_LM_CREATE_CDQ_CDQID_MASK:			Mask to get CNTLID field from Create
+ *							response in completion dword0
+ * @NVME_LM_DELETE_CDQ_CDQID_SHIFT:			Shift amount to set CDQID field for deletion
+ * @NVME_LM_DELETE_CDQ_CDQID_MASK:			Mask to set CDQID field for deletion
+ */
+enum nvme_lm_cdq_fields {
+	NVME_LM_CREATE_CDQ				= 0,
+	NVME_LM_DELETE_CDQ				= 1,
```

Comment:
Use the infix `SEL`, e.g.`NVME_LM_SEL_CREATE_CDQ` . I prefer to keep the naming structure in sync with the spec. Having variations and interpretation is making it really hard to find the spot in the spec.